### ISSUE TITLE
Error while creating Course

### ISSUE THREAD

**Issue by martinvivek** [frumbert/moodle-course_meta#5]
The below error is arising during creating the course.
Notice: Undefined property: stdClass::$id in /course_meta/profile/lib.php in line 390
While editing the course there is no error. everything workign as expected. any suggestions to avoid the error while creating course?

**Comment by frumbert** [comment]
What version of Moodle are you using?
What branch of the plugin are you using?
What web server are you using?
What database system & version are you using?
Does your database grant CREATE VIEW and SHOW VIEW permissions on the moodle db user?

**Comment by martinvivek** [comment]
moodle version - Moodle 3.0.2
Branch - Moodle 2.8.7-MultiSelect-MetaFilter
Webserver - IIS 7
Database - Mysql 5.6
DB user has full access
PHP - 5.6

its not a single error, if we created 3 custom field then 3 error is arising.

**Comment by martinvivek** [comment]
everything is working as expected except that error. it may the issue with the moodle version? i have selected wrongly it seems.

**Comment by frumbert** [comment]
yeah. the plugin adds code to the core routines, so it's tied to the particular moodle build. effectively you're taking a copy of the course edit pages for that particular version, extending them with the extra fields, and then overriding the built-in page using $CFG->customscripts , so if anything in the way courses are configured changes from one moodle version to the next, this page needs re-doing. I haven't done a moodle 3 branch yet, will do soon

**Comment by martinvivek** [comment]
I done the same thing what you have been stated, still the error arises. I am not a high level programmer to check all any how i tried and failed. let me wait until you done it. Thanks for the prompt response.

**Comment by shelyago** [comment]
I'd be also very grateful for the 3.x.x branch. 

I have Moodle 3.0.2 too, running on localhost.
I have managed to install the Course catalogue filter only (which can be changed in Filters, correspondingly), and with it's help the page of course creation doesn't load at all. :) No page, no error, no problem.

Eagerly waiting for your updates!

And thanks for the great job!
